crates/astria-cli/src/commands/rollup.rs

```diff
@@ -0,0 +1,370 @@
+/// * If the arguments cannot be serialized to yaml
+/// * If the yaml cannot be written to the file
+pub(crate) fn create_config(args: &ConfigCreateArgs) -> eyre::Result<()> {
+    // create rollup from args
+    let rollup = Rollup::try_from(args)?;
+    let filename = rollup.deployment_config.get_filename();
+
+    // create file config file
+    let mut output = File::create(&filename)?;
+
+    // write args as yaml
+    let yaml_str: String = rollup.try_into()?;
+    write!(output, "{yaml_str}")?;
+
+    println!("Created rollup config file {filename}");
+
+    Ok(())
+}
+
+/// Deletes a config file
+///
+/// # Arguments
+///
+/// * `args` - The arguments passed to the command
+///
+/// # Errors
+///
+/// * If the config file cannot be deleted
+pub(crate) fn delete_config(args: &ConfigDeleteArgs) -> eyre::Result<()> {
+    let path = PathBuf::from(args.config_path.clone());
+    std::fs::remove_file(path).wrap_err("could not delete the config file")?;
+
+    println!("Deleted rollup config file {}", args.config_path);
+
+    Ok(())
+}
+
+pub(crate) fn edit_config(args: &ConfigEditArgs) -> eyre::Result<()> {
+    // get file contents
+    let path = PathBuf::from(&args.config_path);
+    let mut file = File::open(&path)?;
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)?;
+
+    let mut yaml_value: serde_yaml::Value = serde_yaml::from_str(&contents)?;
+    update_yaml_value(&mut yaml_value, &args.key, &args.value)?;
+
+    // Write the updated YAML back to the file
+    let updated_yaml = serde_yaml::to_string(&yaml_value)?;
+    let mut file = File::create(&path)?;
+    file.write_all(updated_yaml.as_bytes())?;
+
+    Ok(())
+}
+
+/// Creates a deployment from a config file
+///
+/// # Arguments
+///
+/// * `args` - The arguments passed to the command
+///
+/// # Errors
+///
+///
+/// * If the config file cannot be opened
+/// * If the config file cannot be deserialized
+/// * If the deployment cannot be created
+/// * If the helm command fails
+pub(crate) fn create_deployment(args: &DeploymentCreateArgs) -> eyre::Result<()> {
+    let path = PathBuf::from(args.config_path.clone());
+    let file = File::open(path)?;
+    let rollup: Rollup = serde_yaml::from_reader(file)?;
+
+    let chart_path = if let Some(chart_path) = &args.chart_path {
+        chart_path.to_string()
+    } else {
+        EVM_ROLLUP_CHART_URL.to_string()
+    };
+
+    // call `helm install` with appropriate args.
+    // setting values via the generated config file.
+    let helm = helm_from_env();
+    let mut cmd = Command::new(helm.clone());
+    cmd.arg("install")
+        .arg("--debug")
+        .arg("--values")
+        .arg(rollup.deployment_config.get_filename())
+        .arg("--set")
+        // FIXME - is this the best place to set disable_finalization to true?
+        //  bc we probably don't want it in the config file for them to change right now
+        .arg("config.rollup.disableFinalization=true")
```

Comment:
which component is this targeting? i think we can remove this

Reply:
It's for the Conductor. The helm chart has `false` as default, but we want to disable finalization right now to avoid using Celestia to simplify this dem aiui

Reply:
This will also have to be updated to accommodate the execution_commit_level = SoftOnly at some point